panic: runtime error: index out of range [1] with length 1

```
☁  smali2java [master] ./smali2java -path_to_smali=/Userssmali/
.field public transient c:Z
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
```

```
☁  smali2java [master] ./smali2java -path_to_smali=/Users/smali_classes2/
panic: runtime error: slice bounds out of range [:-1]

```


[smali_src.zip](https://github.com/AlexeySoshin/smali2java/files/4148947/smali_src.zip)

--- a/java/modifiers.go
+++ b/java/modifiers.go
@@ -3,6 +3,7 @@ package java
 var Modifiers = map[string]bool{"public": true, "private": true, "protected": true}
 
 const (
-	Final    = "final"
-	Volatile = "volatile"
+	Final     = "final"
+	Volatile  = "volatile"
+	Transient = "transient"
 )

--- a/parser/field.go
+++ b/parser/field.go
@@ -14,6 +14,7 @@ type FieldParser struct {
 	synthetic bool
 	final     bool
 	volatile  bool
+	transient bool
 }
 
 func (p *FieldParser) Parse(javaFile *JavaFile, currentLine Line) error {
@@ -55,11 +56,17 @@ func (p *FieldParser) Parse(javaFile *JavaFile, currentLine Line) error {
 		memberAndClassIndex++
 	}
 
+	if currentLine[memberAndClassIndex] == smali.Transient {
+		p.transient = true
+		memberAndClassIndex++
+	}
+
 	memberAndClass = strings.Split(currentLine[memberAndClassIndex], ":")
 
 	if len(memberAndClass) < 2 {
 		fmt.Println(currentLine)
 	}
+
 	className := GetClassName(memberAndClass[1])
 	memberName := memberAndClass[0]
 	var line []string
@@ -78,6 +85,10 @@ func (p *FieldParser) Parse(javaFile *JavaFile, currentLine Line) error {
 
 	if p.volatile {
 		line = append(line, java.Volatile)
+	}
+
+	if p.transient {
+		line = append(line, java.Transient)
 	}
 
 	line = append(line, className, memberName, ";")

--- a/smali/literals.go
+++ b/smali/literals.go
@@ -34,4 +34,5 @@ const (
 	IPutObject       = "iput-object"
 	IGetObject       = "iget-object"
 	Volatile         = "volatile"
+	Transient        = "transient"
 )

